clarifaipyspark/dataset.py

```diff
@@ -0,0 +1,451 @@
+
+    if df_type not in ('raw', 'url', 'file_path'):
+      raise UserError('Invalid csv type, it should be raw, url or file_path')
+
+    if df_type == 'raw' and input_type != 'text':
+      raise UserError('Only text input type is supported for raw csv type')
+
+    if not isinstance(dataframe, SparkDataFrame):
+      raise UserError('dataframe should be a Spark DataFrame')
+
+    chunk_size = min(128, chunk_size)
+    input_obj = input_obj = Inputs(user_id=self.user_id, app_id=self.app_id)
+    input_protos = self._get_inputs_from_dataframe(
+        dataframe=dataframe,
+        df_type=df_type,
+        input_type=input_type,
+        dataset_id=self.dataset_id,
+        labels=labels)
+    return (input_obj._bulk_upload(inputs=input_protos, chunk_size=chunk_size))
+
+  def upload_dataset_from_dataloader(self,
+                                     task: str,
+                                     split: str,
+                                     module_dir: str = None,
+                                     chunk_size: int = 128) -> None:
+    """Uploads dataset using a dataloader function for custom formats.
+
+    Args:
+        task (str): task type(text_clf, visual-classification, visual_detection, visual_segmentation, visual-captioning).
+        split (str): split type(train, test, val).
+        module_dir (str): path to the module directory.
+        chunk_size (int): chunk size for concurrent upload of inputs and annotations.
+
+    Example: TODO
+    """
+    self.upload_dataset(task=task, split=split, module_dir=module_dir, chunk_size=chunk_size)
+
+  def upload_dataset_from_table(self,
+                                table_path: str,
+                                input_type: str,
+                                table_type: str,
+                                labels: bool,
+                                chunk_size: int = 128) -> None:
+    """upload dataset to clarifai app from spark tables.
+
+    Args:
+        table_path (str): path of the table to be uploaded.
+        input_type (str): Input type of the dataset whether (Image, text).
+        table_type (str): Type of the table contents (url, raw, filepath).
+        labels (bool): Give True if labels column present in dataset else False.
+        chunk_size (int): chunk size for concurrent upload of inputs and annotations.
+      
+    Example: TODO
+    """
+    spark = SparkSession.builder.appName('Clarifai-spark').getOrCreate()
+    tempdf = spark.read.format("delta").load(table_path)
+    self.upload_dataset_from_dataframe(
+        dataframe=tempdf,
+        input_type=input_type,
+        df_type=table_type,
+        labels=labels,
+        chunk_size=chunk_size)
+
+  def list_inputs(self, per_page: int = None, input_type: str = None) -> Generator[Input, None, None]:
+    """Lists all the inputs from the app.
+
+    Args:
+        per_page (str): No of response of inputs per page.
+        input_type (str): Input type that needs to be displayed (text,image)
```

Comment:
if input_type is limited to image, text we could add
```python
if input_type not in ('image', 'text'):
      raise UserError('Invalid input type, it should be image or text')
```

Reply:
Modified.